src/freenet/support/io/Fallocate.java

```diff
@@ -111,4 +118,28 @@ private static int getDescriptor(FileDescriptor descriptor) {
       throw new UnsupportedOperationException("unsupported FileDescriptor implementation", e);
     }
   }
+
+  public static void legacyFill(FileChannel fc, long newLength, long offset, boolean starting) throws IOException {
+    MersenneTwister mt = new MersenneTwister();
+    byte[] b = new byte[4096];
+    ByteBuffer bb = ByteBuffer.wrap(b);
+    int x = 0;
+    while (offset < newLength) {
+      bb.rewind();
+      mt.nextBytes(b);
+      offset += fc.write(bb, offset);
+      if (offset % (1024 * 1024 * 1024L) == 0) {
+        mt = new MersenneTwister();
+        if (starting) {
+          WrapperManager.signalStarting( (int) MINUTES.toMillis(5));
+          if (x++ % 32 == 0)
+            System.err.println(
+                "Preallocating space : "
+                + offset
+                + "/"
+                + newLength);
```

Comment:
toad wanted it to be visible in wrapper.log since that's the only thing users can be expected to see; I can reduce the periodicity if you want... what would be reasonable?